hazelcast-ra/hazelcast-jca/src/test/java/com/hazelcast/jca/HazelcastTransactionImplTest.java

```diff
@@ -0,0 +1,87 @@
+package com.hazelcast.jca;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.test.HazelcastTestSupport;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.Mock;
+import static org.mockito.Mockito.*;
+
+import javax.annotation.Resource;
+import javax.resource.ResourceException;
+import javax.resource.spi.ConnectionEvent;
+import javax.resource.spi.ConnectionEventListener;
+import javax.resource.spi.ManagedConnection;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by mustafaiman on 25/06/15.
+ */
+public class HazelcastTransactionImplTest extends HazelcastTestSupport{
+
+    HazelcastInstance instance;
+    HazelcastTransaction transaction;
+    ManagedConnection managedConnection;
+    ConnectionEventListener mockEventListener;
+
+    private boolean transactionStartedFlag;
+    private boolean transactionCommittedFlag;
+    private boolean transactionRolledbackFlag;
+    private boolean connectionErrorFlag;
+
+
+    @Before
+    public void setup() throws ResourceException {
+        instance = createHazelcastInstance();
+        ResourceAdapterImpl resourceAdapter = new ResourceAdapterImpl();
+        resourceAdapter.setHazelcastInstance(instance);
+        ManagedConnectionFactoryImpl managedConnectionFactory = new ManagedConnectionFactoryImpl();
+        managedConnectionFactory.setResourceAdapter(resourceAdapter);
+        managedConnectionFactory.setConnectionTracingEvents(null);
+        mockEventListener = mock(ConnectionEventListener.class);
+        managedConnection = managedConnectionFactory.createManagedConnection(null, null);
+        managedConnection.addConnectionEventListener(mockEventListener);
+        transaction = new HazelcastTransactionImpl(managedConnectionFactory, (ManagedConnectionImpl) managedConnection);
+
+
+    }
+
+    @Test
+    @Ignore
```

Comment:
This is because ManagedConnectionImpl class never call its listener's localTransactionStarted method because of hardcoded logic. I thought it should call and asked @gurbuzali about it. He was cautious about changing the logic. I left the test there because I think it should be implemented but existing code would not pass the test.